bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/ShellyBindingConstants.java

```diff
@@ -0,0 +1,314 @@
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.shelly.internal;
+
+import java.util.Collections;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+
+/**
+ * The {@link ShellyBindingConstants} class defines common constants, which are
+ * used across the whole binding.
+ *
+ * @author Markus Michels - Initial contribution
+ */
+@NonNullByDefault
+public class ShellyBindingConstants {
+
+    public static final String            VENDOR                           = "Shelly";
+    public static final String            BINDING_ID                       = "shelly";
+
+    // Type names
+    public static final String            THING_TYPE_SHELLY1_STR           = "shelly1";
+    public static final String            THING_TYPE_SHELLY1PN_STR         = "shelly1pm";
+    public static final String            THING_TYPE_SHELLYEM_STR          = "shellyem";
+    public static final String            THING_TYPE_SHELLY2_PREFIX        = "shellyswitch";
+    public static final String            THING_TYPE_SHELLY2_RELAY_STR     = "shelly2-relay";
+    public static final String            THING_TYPE_SHELLY2_ROLLER_STR    = "shelly2-roller";
+    public static final String            THING_TYPE_SHELLY25_PREFIX       = "shellyswitch25";
+    public static final String            THING_TYPE_SHELLY25_RELAY_STR    = "shelly25-relay";
+    public static final String            THING_TYPE_SHELLY25_ROLLER_STR   = "shelly25-roller";
+    public static final String            THING_TYPE_SHELLY4PRO_STR        = "shelly4pro";
+    public static final String            THING_TYPE_SHELLYPLUG_STR        = "shellyplug";
+    public static final String            THING_TYPE_SHELLYPLUGS_STR       = "shellyplugs";
+    public static final String            THING_TYPE_SHELLYDIMMER_STR      = "shellydimmer";
+    public static final String            THING_TYPE_SHELLYBULB_STR        = "shellybulb";
+    public static final String            THING_TYPE_SHELLYRGBW2_PREFIX    = "shellyrgbw2";
+    public static final String            THING_TYPE_SHELLYRGBW2_COLOR_STR = "shellyrgbw2-color";
+    public static final String            THING_TYPE_SHELLYRGBW2_WHITE_STR = "shellyrgbw2-white";
+    public static final String            THING_TYPE_SHELLYHT_STR          = "shellyht";
+    public static final String            THING_TYPE_SHELLYSMOKE_STR       = "shellysmoke";
+    public static final String            THING_TYPE_SHELLYFLOOD_STR       = "shellyflood";
+    public static final String            THING_TYPE_SHELLYEYE_STR         = "shellyseye";
+    public static final String            THING_TYPE_SHELLYSENSE_STR       = "shellysense";
+    public static final String            THING_TYPE_SHELLYUNKNOWN_STR     = "shellydevice";
```

Comment:
You might want to change the constant to THING_TYPE_DEVICE_STR.

Reply:
My suggestion was to simply use "device" (and not "shellydevice").
In general, it would be more straight forward to change all thing types to a string that does not have the  "shelly" prefix as this is really just repetitive, since the type always has the binding id in front of it. "shelly:device" simply makes much more sense then "shelly:shellydevice", wdyt?

Reply:
This means
- changing definitions
- changing xml
- changing code
- changing doc

Not the best idea if we want to release tomorrow and we have up to 60 users, which might get pissed, because they need to redo all their thing definitions or linkage.

Reply:
Ok, understood. So take it as a recommendation for future bindings then :-)